### PR TITLE
Fix bitcast assertion in replaceLoadStoreInst

### PR DESCRIPTION
**Pull request by xuechen417**
- The new test ObjStorageBuffer_TestStoreVectorArray.spvasm has an array type constant value to store to a vector which triggered the bitcast assertion.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit a6d988e51c4e6b3e89c697b38692bbe3a01f6f90
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 85170384d0ee5447ac267d478f7a21bc036ea71c</li>
<li>SPVGEN: dev 704f4adddb6867d633431c2c9c2c0bfa6d294ce2</li>
<li>XGL: dev 9c4fe791366e730131a0d99a7519855068f216ef</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev c57487b76215ee74b1038039f47b210a28cad65e</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 961/961 (100.0%)</li>
<li>Failed: 0/961 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103143/103143 (100.0%)</li>
<li>Failed: 0/103143 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by JaxLinAMD** [comment]
retest this please

File: lgc/patch/PatchBufferOp.cpp [hunk 1]
```diff
@@ -1400,14 +1400,26 @@ Value *PatchBufferOp::replaceLoadStore(Instruction &inst) {
   Value *storeValue = nullptr;
   if (!isLoad) {
     storeValue = storeInst->getValueOperand();
-
-    if (storeValue->getType()->isPointerTy()) {
+    Type *storeTy = storeValue->getType();
+    if (storeTy->isPointerTy()) {
```
Comment (amdrexu) [inline lgc/patch/PatchBufferOp.cpp:1404]: I think the logic is expected to change like this:

```
if ((storeTy->isArrayTy()) {
  // Your new codes
} else {
  // Old codes
  if (storeTy->isPointerTy())
    XXX;

  YYY;
}
```

Comment (xuechen417) [reply lgc/patch/PatchBufferOp.cpp]: Fixed.

**Comment by amdvlk-admin** [comment]
### Test summary for commit d1f750876dae52e0a2e2a3d300b18cd4f72f882f
<details open><summary>Driver commits used in build</summary><ul>

<li>CWPACK: amd-master b601c88aeca7a7b08becb3d32709de383c8ee428</li>
<li>METROHASH: amd-master 2b6fee002db6cc92345b02aeee963ebaaf4c0e2f</li>
<li>PAL: dev 85170384d0ee5447ac267d478f7a21bc036ea71c</li>
<li>SPVGEN: dev 704f4adddb6867d633431c2c9c2c0bfa6d294ce2</li>
<li>XGL: dev 9c4fe791366e730131a0d99a7519855068f216ef</li>
<li>LLVM-PROJECT: amd-gfx-gpuopen-dev 3a1103f746f4d782271bd32d67274d6260c5dc7f</li>
</ul></details>

<details open><summary>ShaderDB test</summary><ul>

<li>Passed: 960/960 (100.0%)</li>
<li>Failed: 0/960 (0.0%)</li>
</ul></details>

<details open><summary>CTS tests (built with branch Vulkan-CTS-1.2.0.2-fetch-sources-fix)</summary><ul>


<details open><summary>Ubuntu 18.04, Ellesmere</summary><ul>

<li>Passed: 103143/103143 (100.0%)</li>
<li>Failed: 0/103143 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 7.6, Vega10</summary><ul>

<li>Passed: 103199/103199 (100.0%)</li>
<li>Failed: 0/103199 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 18.04, Navi10</summary><ul>

<li>Passed: 102548/102549 (100.0%)</li>
<li><details><summary>Failed: 1/102549 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.transform_feedback.fuzz.random_geometry.all_unordered_and_missing.15
Stack trace: Timeout or crahsed
```
</p></details></li>


</ul></details>


</ul></details>

**Pull request merged**